### ISSUE TITLE
[Assistive tech and device support] Alert and status messages aren't announced without receiving focus. (11.22.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#52076]
#### VFS team name
DGIB - My Education Benefits

#### VFS product name
Enrollment verification

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Alert and status messages aren't announced without receiving focus.

### Issue Details
When a user successfully verifies enrollment for both months, they are brought back to the Post-9/11 GI Bill enrollment verifications page, and a green success alert appears beneath the intro content. For screen reader users, focus is set to the top of the page, meaning they won't find that success message unless they know to explore the page to look for it. Since the alert doesn't have a heading, it's not particularly discoverable for a screen reader user either -- they'll have to start listening to content from the H1, sit through the intro paragraph, and then finally hear that their verification was successful.

I'm logging this as a launch-blocking issue, because screen reader users will likely have a very difficult time figuring out whether the action they took just worked and may erroneously look for a way to resubmit assuming that it didn't work.

Note that this isn't an issue when the verification workflow ends in payments being paused. See the recommendations section below for details.

### Link, screenshot or steps to recreate
Steps to reproduce:
1. With test user 301, complete the enrollment verification workflow up to the point of the final submission.
2. Activate a screen reader (recommend VoiceOver on Mac or NVDA on Windows).
3. Press submit.
4. Note that you're navigated back to the enrollment verifications page, but no announcement is made about your action being successful.

### VA.gov Experience Standard
[Category Number 11, Issue Number 22](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 4.1.3_AA

### Platform Recommendation
This is partly related to a choice made in the design system about the [alert component](https://design.va.gov/components/alert). For red error alerts, the alert has `role="alert" aria-live="assertive"` attributes added to it. This ensures that critical information gets announced immediately to screen reader users, before other content on the page. The other alert styles (info, success, warning, continue) don't have those attributes added. In a lot of use cases for the alert component that makes sense -- we probably don't want to interrupt the normal flow of information for something that's not critical. But in this case the assumptions behind the component break down.

Two possible solutions:
1. Wrap the `va-alert` component with that success message in a div with `role="alert" aria-live="assertive"`. Note that I haven't tested to verify that this will behave as expected, but I think it should announce the success message as soon as you land back on the page after submission.
2. If displaying that success message, set focus to content in the alert -- preferably setting focus to an `h2` heading at the top of the alert.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by cesaresantiago** [comment]
Copying over @nathanielbuck 's comment from Jira:

"I had a bit of trouble implementing this. We don't control (and can't override) the markup that gets generated by VaRadio. So, while I was able to reorder content, I was not able to add the aria-describedby label in the exact spot requested. In my testing with VoiceOver, it didn't seem that moving the alert made it easier to access the alert. That content seemed to be skipped over still."

Since we are limited by the capabilities of the VaRadio component of the form system, we will not be able to address the voiceover issue. If this is still an issue, it needs to be addressed on the form system.

cc: @btruongafs , @shiragoodman

**Comment by btruongafs** [comment]
This issue has been resolved and deployed to staging.

**Comment by shiragoodman** [comment]
hi @btruongafs @cesaresantiago @nathanielbuck - I wanted to follow up on this one bc per the comment from last week, you're limited by the capabilities of the VaRadio component, and now this week it's resolved and deployed to staging. @btruongafs did you comment on the correct ticket? Or did you find a way to address the issue within the component?

As stated before, we prefer tickets be kept open if they're not resolved. If this issue has not been resolved and you're marking it closed because you're not responsible for the fix, then we will need to reopen the ticket and add the team that is responsible for the fix (likely Design System team).

Please let me know.

**Comment by btruongafs** [comment]
@shiragoodman we implemented the solution provided by @briandeconinck
